alerting/src/main/kotlin/org/opensearch/alerting/util/DocLevelMonitorQueries.kt

```diff
@@ -389,6 +461,62 @@ class DocLevelMonitorQueries(private val client: Client, private val clusterServ
         return Pair(updateMappingResponse, targetQueryIndex)
     }
 
+    private fun mergeConflictingFields(oldField: Map<String, Any>, newField: Map<String, Any>): Map<String, Any> {
+        val mergedField = mutableMapOf<String, Any>()
+        oldField.entries.forEach {
+            if (newField.containsKey(it.key)) {
+                if (it.value is Map<*, *> && newField[it.key] is Map<*, *>) {
+                    mergedField[it.key] =
+                        mergeConflictingFields(it.value as Map<String, Any>, newField[it.key] as Map<String, Any>)
+                } else {
+                    mergedField[it.key] = it.value
+                }
+            } else {
+                mergedField[it.key] = it.value
+            }
+        }
+
+        newField.entries.forEach {
+            if (!oldField.containsKey(it.key)) {
+                mergedField[it.key] = it.value
+            }
+        }
+        return mergedField
+    }
+
+    fun getAllConflictingFields(clusterState: ClusterState, concreteIndices: List<String>): Set<String> {
+        val conflictingFields = mutableSetOf<String>()
+        val allFlattenPaths = mutableMapOf<String, MutableMap<String, Any>>()
```

Comment:
does the below flatten logic handle both the nested and non-nested type objects correctly?

what would be the difference after flattening into string in the following 2 cases 
```
"nested_field": {
                      "type": "nested",
                      "properties": {
                        "test1": {
                          "type": "keyword"
                        }
                      }
                      
```

and 


```
     "nested_field": {
                      
                      "properties": {
                        "test1": {
                          "type": "keyword"
                        }
                      }

```

Reply:
`nested fields` dont work with `query string queries`. https://stackoverflow.com/questions/69857071/how-can-i-use-query-string-to-match-both-nested-and-non-nested-fields-at-the-sam
but if we pair them with an index containing an object field, it will work. https://github.com/opensearch-project/alerting/pull/1097/files#diff-9b08d53e8cff3d739beb6ba304e4eaf466f08412762c3cd55886a52b722a77f1R636